### PR TITLE
set SyliusShopApiPlugin and SyliusElasticSearchPlugin as not obligatory

### PR DESCRIPTION
**Pull request by kuzyk**

### CODE REVIEW

**Comment by pamil** [comment]
Awesome, would love to see it loosely coupled. The Travis is currently failing because these dependencies should be still included in `require-dev`.

**Comment by pjedrzejewski** [comment]
@kuzyk Nice work, would be great to have this decoupling!

File: node_modules [hunk 1]
```diff
@@ -1 +0,0 @@
-tests/Application/node_modules
```
Comment (lchrusciel) [inline node_modules]: This file should stay

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
+.idea
```
Comment (lchrusciel) [inline .gitignore]: This should be a part of your private composer configuration

File: composer.json [hunk 3]
```diff
@@ -5,11 +5,11 @@
...
     "require": {
         "php": "^7.1",
-
-        "sylius/elastic-search-plugin": "^1.0@alpha",
-        "sylius/shop-api-plugin": "^1.0@beta",
-        "sylius/sylius": "^1.0@dev",
-        "symfony/monolog-bundle": "^3.0"
+        "sylius/sylius": "^1.0@dev"
+    },
+    "suggest": {
```
Comment (lchrusciel) [inline composer.json]: 👍

**Comment by lchrusciel** [comment]
Hello @kuzyk. Sorry for pinging you one more time, but there is a strong need for this fix in the community. Are you willing to rebase to the newest master your changes?

**Comment by psihius** [comment]
This PR isn't relevant any more and being superseded by https://github.com/URBANARA/UrbanaraCatalogPromotionPlugin/pull/39 witch includes changes from this PR plus compatibility with Elastic Plugin and other changes

**Comment by lchrusciel** [comment]
@psihius 👍

**Comment by lchrusciel** [comment]
Thanks @kuzyk for all your effort! :)

**Pull request closed without merging**